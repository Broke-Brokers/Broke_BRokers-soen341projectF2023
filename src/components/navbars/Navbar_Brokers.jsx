import React, { useState, useEffect } from "react";
import { Link } from 'react-router-dom';
import { Button } from './../Button';
import { ErrorBoundary } from "react-error-boundary";

import './Navbar.css';

function Navbar_Brokers() {
  
    const [click, setClick] = useState(false);
    const[button,setButton]= useState(true);
    const handleClick = ()=>setClick(!click);
    const closeMobileMenu=() => setClick(false); 

const showButton = ()=>{
    if (window.innerWidth<=960) {
        setButton(false);
    } else {
        setButton(true);
    }
};

useEffect(()=>{showButton();},[]);
window.addEventListener('resize', showButton);
  return (
    <>
    <ErrorBoundary fallback={<div>Something went wrong</div>}>
   <nav className="navbar">
    <div className="navbar-container"> 

        <Link to= "/"className="navbar-logo" onClick={closeMobileMenu}>
            BrokeBroker
        </Link>
        <div className="menu-icon" onClick={handleClick}>
            <i className={click ? 'fas fa-times' : 'fas fa-bars'}/>
        </div>

        


        <ul className={click? 'nav-menu active' : 'nav-menu'}>
          
        <li className='nav-item'>
          <Link to='/search' className='nav-links nav-button' onClick={closeMobileMenu}>
            Search for properties
              </Link>
              </li>
                
          <li className='nav-item'>
            <Link to='/clients' className='nav-links' onClick={closeMobileMenu}>
               Clients
            </Link>
          </li>
          <li className='nav-item'>
            <Link to='/offers' className='nav-links' onClick={closeMobileMenu}>
               Offers
            </Link>
          </li>
          

              <li className='nav-item'>
          <Link to='/profile' className='nav-links ' onClick={closeMobileMenu}>
            My Profile
              </Link>
              </li>

        </ul>
      {/*We will need to add  a signout */}
        {button && <Button buttonStyle='btn--outline'>SIGN IN</Button>}


        
    </div>
   </nav>
   </ErrorBoundary>
   </>
  );
}

export default Navbar_Brokers;
