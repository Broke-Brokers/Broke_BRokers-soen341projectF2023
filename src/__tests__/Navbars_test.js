import React, { useState, useEffect } from "react";
import { Link } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar_Brokers from '../components/navbars/Navbar_Brokers';
import Navbar_Homebuyer from '../components/navbars/Navbar_Homebuyer';
import Navbar_SysAdmin from "../components/navbars/Navbar_SysAdmin";

it('performs snapshot testing', () => {
    const tree = renderer.create(<Navbar_Brokers />).toJSON();
    expect(tree).toMatchSnapshot();
    });
it('performs snapshot testing', () => {
        const tree = renderer.create(<Navbar_Homebuyer />).toJSON();
        expect(tree).toMatchSnapshot();
        });
it('performs snapshot testing', () => {
            const tree = renderer.create(<Navbar_SysAdmin />).toJSON();
            expect(tree).toMatchSnapshot();
            });