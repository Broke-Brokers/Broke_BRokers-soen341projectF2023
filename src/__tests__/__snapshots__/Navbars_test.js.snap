// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`performs snapshot testing 1`] = `
<div>
  Something went wrong
</div>
`;

exports[`performs snapshot testing 2`] = `
<div>
  Something went wrong
</div>
`;

exports[`performs snapshot testing 3`] = `
<div>
  Something went wrong
</div>
`;
